spec/api/sul_bib/publications_api_spec.rb

```diff
@@ -269,49 +269,87 @@ def validate_authorship(pub_hash, submission)
-        identifier: [{
-          type: 'SULPubId',
-          id: 'n',
-          url: 'm'
-        }],
-        authorship: [{
-          sul_author_id: author.id,
-          status: 'denied',
-          visibility: 'public',
-          featured: true
-        }]
-      }.to_json
-      put "/publications/#{publication.id}", json_with_sul_pub_id, headers
-      expect(result['identifier'].count { |x| x['type'] == 'SULPubId' }).to eq(1)
-      expect(result['identifier'][0]['id']).not_to eq('n')
-    end
 
-    it 'updates an existing pub' do
-      post '/publications', with_isbn_hash.to_json, headers
-      id = Publication.last.id
-      put "/publications/#{id}", with_isbn_changed_doi.to_json, headers
-      expect(result['identifier'].size).to eq(3)
-      expect(result['identifier']).to include(
-        a_hash_including('type' => 'isbn', 'id' => '1177188188181'),
-        a_hash_including('type' => 'doi', 'url' => '18819910019-updated'),
-        a_hash_including('type' => 'SULPubId', 'id' => id.to_s, 'url' => "#{Settings.SULPUB_ID.PUB_URI}/#{id}")
-      )
+    context 'successfully' do
+      after { expect(response.status).to eq(200) }
+
+      it 'does not duplicate SULPubIDs' do
+        json_with_sul_pub_id = {
+          type: 'book',
+          identifier: [{
+            type: 'SULPubId',
+            id: 'n',
+            url: 'm'
+          }],
+          authorship: [{
+            sul_author_id: author.id,
+            status: 'denied',
+            visibility: 'public',
+            featured: true
+          }]
+        }.to_json
+        put "/publications/#{publication.id}", json_with_sul_pub_id, headers
+        expect(result['identifier'].count { |x| x['type'] == 'SULPubId' }).to eq(1)
+        expect(result['identifier'][0]['id']).not_to eq('n')
+      end
+
+      it 'updates existing pub' do
+        post '/publications', with_isbn_hash.to_json, headers
+        id = Publication.last.id
+        put "/publications/#{id}", with_isbn_changed_doi.to_json, headers
+        expect(result['identifier'].size).to eq(3)
+        expect(result['identifier']).to include(
+          a_hash_including('type' => 'isbn', 'id' => '1177188188181'),
+          a_hash_including('type' => 'doi', 'url' => '18819910019-updated'),
+          a_hash_including('type' => 'SULPubId', 'url' => "#{Settings.SULPUB_ID.PUB_URI}/#{id}", 'id' => id.to_s)
+        )
+      end
+
+      it 'deletes an identifier from the db if it is not in the incoming json' do
+        post '/publications', with_isbn_hash.to_json, headers
+        id = Publication.last.id
+        put "/publications/#{id}", with_isbn_deleted_doi.to_json, headers
+        expect(result['identifier'].size).to eq(2)
+        expect(result['identifier']).to include(
+          a_hash_including('type' => 'isbn', 'id' => '1177188188181'),
+          a_hash_including('type' => 'SULPubId', 'url' => "#{Settings.SULPUB_ID.PUB_URI}/#{id}", 'id' => id.to_s)
+        )
+      end
```

Comment:
Does it really behave this way?  Or does something, somewhere in this PR make it behave this way?  Is it OK for a PUT on a publication to delete existing identifiers?  I wonder if this depends on the provenance of the publication?  AFAIK, we do not allow any changes to publication or publication-identifiers unless the provenance is a manual submission ('cap' provenance).  This spec indicates otherwise.